plugins/inventory/icinga2.py

```diff
@@ -232,11 +251,9 @@ def _convert_inv(self, json_data):
                                         host_attrs['state'])
             self.inventory.set_variable(host_name, 'state_type',
                                         host_attrs['state_type'])
-            if host_attrs.get('vars') is not None:
-                if host_attrs['vars'].get('ansible_user') is not None:
-                    self.inventory.set_variable(host_name, 'ansible_user', host_attrs['vars'].get('ansible_user'))
-                if host_attrs['vars'].get('ansible_port') is not None:
-                    self.inventory.set_variable(host_name, 'ansible_port', host_attrs['vars'].get('ansible_port'))
+            # Adds all attributes to a host variable 'icinga2'
+            self.inventory.set_variable(host_name, 'icinga2', host_attrs)
```

Comment:
It might be better not to set this as a variable for every host - users might have already used `icinga2` for something else. But it should definitely be provided to `_apply_constructable`.

Users can then still use `compose` to make `icinga2` visible in playbooks/roles:
```
compose:
  icinga2: icinga2
```
(Or give it another name, or decide to not to expose it.)

Reply:
Thank you! Honestly, I don't understand how I could include the additional information (attributes) that icinga2 will provide if I don't add it as a variable. If I change the name of the variable, will it be enough? Is there any convention for these variables? If I use for example `icinga2_attributes`?

Reply:
I've added a suggestion how this could work.

About the naming, that's a good question. I'm not sure there is a convention. If this is a variable that's automatically exposed, it should have a more explicit name (that reduces the probability for collisions). But not automatically exposing it is even better IMO, especially for already existing inventory plugins.